ui/src/app/components/runtime-configuration-panel/create-panel.tsx

```diff
@@ -10,64 +10,82 @@ import { FlexRow } from 'app/components/flex';
 import { RuntimeSummary } from 'app/components/runtime-summary';
 import { ComputeType } from 'app/utils/machines';
 import { runtimePresets } from 'app/utils/runtime-presets';
-import {
-  AnalysisConfig,
-  PanelContent,
-  RuntimeStatusRequest,
-} from 'app/utils/runtime-utils';
+import { AnalysisConfig, PanelContent } from 'app/utils/runtime-utils';
 
 export interface CreatePanelProps {
-  profile: Profile;
-  setPanelContent: (panelContent: PanelContent) => void;
-  workspace: Workspace;
   analysisConfig: AnalysisConfig;
   creatorFreeCreditsRemaining: number;
+  onClose: () => void;
+  onPause: () => void;
+  onResume: () => void;
+  profile: Profile;
+  requestAnalysisConfig: (ac: AnalysisConfig) => void;
+  runtimeCanBeCreated: boolean;
   runtimeStatus: RuntimeStatus;
-  setRuntimeStatusRequest: (runtimeStatusRequest: RuntimeStatusRequest) => void;
+  setPanelContent: (panelContent: PanelContent) => void;
+  workspace: Workspace;
 }
 export const CreatePanel = ({
-  profile,
-  setPanelContent,
-  workspace,
   analysisConfig,
   creatorFreeCreditsRemaining,
+  onClose,
+  onPause,
+  onResume,
+  profile,
+  requestAnalysisConfig,
+  runtimeCanBeCreated,
   runtimeStatus,
-  setRuntimeStatusRequest,
```

Comment:
this was only used to define the pause and resume functions, so I thought it made more sense to pass in those functions instead

Reply:
I wonder if it would make sense to make onPause and onResume optional for StartStopEnvironmentButton (the component where they are ultimately used). If so, I think that this would be a future PR.

I wonder because the StartStopEnvironmentButton has 11 different states, and onPause and onResume are both only used in a single state each.

The runtime configuration panel should always have the StartStopEnvironmentButton in the UNKNOWN state. This state uses neither action, so I believe that the versions of onPause and onResume passed in here are never used.

I don't think that this is related to the change that you are doing here, but looking over the code got me curious.

Reply:
Do you mean that Pause/Resume are not relevant for Create Panel because we'd never see it for those runtime states?

Reply:
Yeah, is that correct?

Reply:
yeah, makes sense.  I'll look into this idea.

Reply:
#8023